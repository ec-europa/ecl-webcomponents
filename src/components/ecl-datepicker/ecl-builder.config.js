const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-datepicker/datepicker.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-datepicker-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-datepicker-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-datepicker-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-datepicker-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-datepicker-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
  ],
};
