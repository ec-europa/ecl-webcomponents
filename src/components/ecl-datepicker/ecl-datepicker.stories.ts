export default {
  title: 'Components/form/datepicker',
};

const getArgs = () => {
  return {
    inputId: 'ecl-datepicker',
    defaultValue: '11-10-2021',
    required: true,
    disabled: false,
    invalid: false,
    label: 'Label',
    placeholder: 'DD-MM-YYYY',
    dateFormat: 'DD-MM-YYYY',
    yearRange: 40,
  };
}

const getArgTypes = () => {
  return {
    disabled: {
      type: { name: 'boolean' },
      description: 'Disabled input',
      table: {
        category: 'States',
      },
    },
    required: {
      type: { name: 'boolean' },
      description: 'Required input',
      table: {
        category: 'States',
      },
    },
    invalid: {
      type: { name: 'boolean' },
      description: 'Invalid input',
      table: {
        category: 'States',
      },
    },
    label: {
      type: { name: 'string' },
      description: 'Label',
      table: {
        category: 'Content',
      },
    },
    inputId: {
      name: 'Input id',
      type: { name: 'string' },
      description: 'Input id',
      table: {
        category: 'Content',
      },
    },
    placeholder: {
      type: { name: 'string' },
      description: 'Placeholder text',
      table: {
        category: 'Content',
      },
    },
    defaultValue: {
      name: 'value',
      type: { name: 'string' },
      description: 'Default value of the input',
      table: {
        category: 'Content',
      },
    },
    dateFormat: {
      name: 'format for the date',
      type: { name: 'string' },
      description: 'You can use any valid date format',
      table: {
        category: 'Content',
      },
    },
    yearRange: {
      name: 'range of years to be considered',
      type: { name: 'number' },
      table: {
        category: 'Content',
      },
    },
  };
}

const Template = args =>
  `<ecl-form-group
    label="${args.label}"
    required=${args.required}
    optional-text=" (optional)"
    helper-text="This is the input's helper text."
    invalid=${args.invalid}
    disabled=${args.disabled}
    invalid-text="This is the error message"
    theme="${args.theme}"
  >
    <ecl-datepicker
      theme="${args.theme}"
      required=${args.required}
      disabled=${args.disabled}
      invalid=${args.invalid}
      input-id="${args.inputId}"
      default-value="${args.defaultValue}"
      placeholder="${args.placeholder}"
      date-format="${args.dateFormat}"
      year-range="${args.yearRange}"
    >
    </ecl-datepicker>
  </ecl-form-group>`;

export const Datepicker = Template.bind({});
Datepicker.storyName = 'default';
Datepicker.args = getArgs();
Datepicker.argTypes = getArgTypes();
