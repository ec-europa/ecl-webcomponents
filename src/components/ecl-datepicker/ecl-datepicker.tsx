import { Component, h, Prop, Element, Event, EventEmitter } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";
declare const ECL: any;

@Component({
  tag: 'ecl-datepicker',
  styleUrls: {
    ec: './build/styles/ecl-datepicker-ec.css',
    eu: './build/styles/ecl-datepicker-eu.css',
  },
  shadow: false,
  scoped: false,
})

export class EclDatepicker {
  @Event() valueChanged: EventEmitter<string>;
  @Event() focused: EventEmitter<void>;
  @Event() blurred: EventEmitter<void>;
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() disabled: boolean = false;
  @Prop() required: boolean = false;
  @Prop() placeholder: string = 'DD-MM-YYYY';
  @Prop() inputId: string;
  @Prop() invalid: boolean = false;
  @Prop() type: string;
  @Prop() name: string;
  @Prop() defaultValue: string;
  @Prop() dateFormat: string = 'DD-MM-YYYY';
  @Prop() yearRange: number = 40;

  handleChange(event: Event) {
    const newValue = (event.target as HTMLInputElement).value;
    this.valueChanged.emit(newValue);
  }

  handleFocus() {
    this.focused.emit();
  }

  handleBlur() {
    this.blurred.emit();
  }

  getClass(): string {
    const styleClasses = ['ecl-datepicker', this.styleClass];

    if (this.invalid) {
      styleClasses.push('ecl-datepicker--invalid');
    }

    if (this.disabled) {
      styleClasses.push('ecl-datepicker--disabled');
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    const momentSrc = 'https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.29.1/moment.min.js';

    if (document.querySelector(`script[src="${momentSrc}"]`)) {
      document.querySelector(`script[src="${momentSrc}"]`).remove();
    }

    const moment = document.createElement('script');
    moment.setAttribute('integrity', 'sha512-qTXRIMyZIFb8iQcfjXWCO8+M5Tbc38Qi5WzdPOYZHIlZpzBHG3L3by84BBBOiRGiEb7KKtAOAs5qYdUiZiQNNQ=');
    moment.setAttribute('crossorigin', 'anonymous');
    moment.src = momentSrc;

    moment.onload = () => {
      const src = getAssetPath('./build/scripts/ecl-datepicker-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }

      const script = document.createElement('script');
      script.src = src;
      script.defer = true;
      script.onload = () => {
        const datepicker = new ECL.Datepicker(
          this.el.querySelector('.ecl-datepicker__field'),
          { format: this.dateFormat, yearRange: Number(this.yearRange) }
        );

          datepicker.init();
      };

      document.body.appendChild(script);
    };

    document.body.appendChild(moment);
  }

  render() {
    return (
      <div class={this.getClass()}>
        <input
          class={`ecl-datepicker__field ecl-text-input sc-ecl-text-input-${this.theme} ecl-text-input--s ${this.invalid ? 'ecl-text-input--invalid' : ''}`}
          autocomplete="off"
          data-ecl-datepicker-toggle
          id={this.inputId}
          value={this.defaultValue}
          required={this.required}
          disabled={this.disabled}
          placeholder={this.placeholder}
          onChange={event => this.handleChange(event)}
          onFocus={() => this.handleFocus()}
          onBlur={() => this.handleBlur()}
        />
        <ecl-icon
          theme={this.theme}
          styleClass={`ecl-datepicker__icon sc-ecl-datepicker-${this.theme}`}
          icon="calendar"
          size="s"
        >  
        </ecl-icon>
      </div>
    );
  }
}