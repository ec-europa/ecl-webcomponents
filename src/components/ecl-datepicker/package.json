{
  "name": "@ecl/web-components-datepicker",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.9.0",
  "description": "ECL Datepicker",
  "main": "ecl-datepicker.ts",
  "private": true,
  "scripts": {
    "build:scripts": "cross-env ecl-builder scripts",
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.9.1",
    "@ecl/theme-ec": "3.9.1",
    "@ecl/theme-eu": "3.9.1",
    "@ecl/vanilla-component-datepicker": "3.9.1",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
