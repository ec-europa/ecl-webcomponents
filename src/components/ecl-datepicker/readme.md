# ecl-datepicker



<!-- Auto Generated Below -->


## Properties

| Property       | Attribute       | Description | Type      | Default        |
| -------------- | --------------- | ----------- | --------- | -------------- |
| `dateFormat`   | `date-format`   |             | `string`  | `'DD-MM-YYYY'` |
| `defaultValue` | `default-value` |             | `string`  | `undefined`    |
| `disabled`     | `disabled`      |             | `boolean` | `false`        |
| `hasChanged`   | `has-changed`   |             | `boolean` | `false`        |
| `inputId`      | `input-id`      |             | `string`  | `undefined`    |
| `invalid`      | `invalid`       |             | `boolean` | `false`        |
| `isFocused`    | `is-focused`    |             | `boolean` | `false`        |
| `name`         | `name`          |             | `string`  | `undefined`    |
| `placeholder`  | `placeholder`   |             | `string`  | `'DD-MM-YYYY'` |
| `required`     | `required`      |             | `boolean` | `false`        |
| `styleClass`   | `style-class`   |             | `string`  | `undefined`    |
| `theme`        | `theme`         |             | `string`  | `'ec'`         |
| `type`         | `type`          |             | `string`  | `undefined`    |
| `yearRange`    | `year-range`    |             | `number`  | `40`           |


## Events

| Event         | Description | Type                      |
| ------------- | ----------- | ------------------------- |
| `inputBlur`   |             | `CustomEvent<FocusEvent>` |
| `inputChange` |             | `CustomEvent<any>`        |
| `inputFocus`  |             | `CustomEvent<FocusEvent>` |


## Dependencies

### Depends on

- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-datepicker --> ecl-icon
  style ecl-datepicker fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
