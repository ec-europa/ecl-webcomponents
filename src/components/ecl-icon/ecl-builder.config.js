const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  copy: [
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-ec-icons/dist/sprites/icons.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-eu-icons/dist/sprites/icon.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
