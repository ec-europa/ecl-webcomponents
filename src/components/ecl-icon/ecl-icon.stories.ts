import iconPathEc from '@ecl/resources-ec-icons/dist/sprites/icons.svg';
import iconPathEu from '@ecl/resources-eu-icons/dist/sprites/icons.svg';
import iconsAllEc from '@ecl/resources-ec-icons/dist/lists/all.json';

export default {
  title: 'Components/icon',
};

const Template = (args) => `<ecl-icon 
                              icon="${args.icon}"
                              color="${args.color}"
                              size="${args.size}"
                              transform="${args.transform}"
                              path="${args.theme === 'ec' ? iconPathEc : iconPathEu}"
                            ></ecl-icon>`;

export const Icon = Template.bind({});
Icon.storyName = 'default';
Icon.args = {
  theme: 'ec',
  icon: 'close',
  transform: '',
  color: 'default',
  size: '2xl',
};
Icon.argTypes = {
  theme: {
    type: { name: 'select'},
    options: ['ec', 'eu'],
  },
  icon: {
    type: { name: 'select' },
    options: iconsAllEc,
    description: "Add an icon",
  },
  size: {
    type: { name: 'select' },
    options: ['2xs','xs', 's', 'm', 'l', 'xl', '2xl'],
    description: "icon size",
  },
  color: {
    type: { name: 'select' },
    options: ['default', 'primary', 'inverted'],
    description: "icon color",
  },
  transform: {
    type: { name: 'select' },
    description: 'Button icon transform',
    options: [
      'rotate-90',
      'rotate-180',
      'rotate-270',
      'flip-horizontal',
      'flip-vertical',
    ],
  },
};