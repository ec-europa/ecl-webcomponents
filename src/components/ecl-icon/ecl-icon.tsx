import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}