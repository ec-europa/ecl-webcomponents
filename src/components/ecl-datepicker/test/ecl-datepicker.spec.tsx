import { newSpecPage } from '@stencil/core/testing';
import { EclDatepicker } from '../ecl-datepicker';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-datepicker', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclDatepicker, EclIcon],
      html: `<ecl-datepicker
              required
              input-id="ecl-datepicker"
              default-value="11-10-21"
              placeholder="DD-MM-YYYY"
            >
            </ecl-datepicker>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
