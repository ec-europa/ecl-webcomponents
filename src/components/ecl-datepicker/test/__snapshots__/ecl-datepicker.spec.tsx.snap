// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-datepicker renders 1`] = `
<ecl-datepicker default-value="11-10-21" input-id="ecl-datepicker" placeholder="DD-MM-YYYY" required="">
  <div class="ecl-datepicker">
    <input autocomplete="off" class="ecl-datepicker__field ecl-text-input ecl-text-input--s sc-ecl-text-input-ec" data-ecl-datepicker-toggle id="ecl-datepicker" placeholder="DD-MM-YYYY" required="" value="11-10-21">
    <ecl-icon>
      <svg class="ecl-datepicker__icon ecl-icon ecl-icon--s sc-ecl-datepicker-ec">
        <use xlink:href="/build/images/ec/icons.svg#calendar"></use>
      </svg>
    </ecl-icon>
  </div>
</ecl-datepicker>
`;
